crates/flux-middle/src/rty/mod.rs

```diff
@@ -966,6 +966,14 @@ pub enum SortArg {
     BvSize(BvSize),
 }
 
+#[derive(Debug, Clone, Eq, PartialEq, Hash, TyEncodable, TyDecodable)]
+pub enum ConstantInfo {
+    /// An uninterpreted constant
+    Uninterpreted,
+    /// A non-integral constant whose value is specified by the user
+    Interpreted(Expr),
```

Comment:
It may make sense to store the `sort` here because we need it when constructing it for sort checking and then we need the sort when using it.

```suggestion
    Interpreted(Expr, Sort),
```